hr_contract_multi_jobs/models/hr_contract.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (C) 2014 Savoir-faire Linux. All Rights Reserved.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class HrContract(models.Model):
+    _inherit = 'hr.contract'
+
+    contract_job_ids = fields.One2many('hr.contract.job',
+                                       'contract_id',
+                                       string='Jobs')
+
+    # Modify the job_id field so that it points to the main job
+    job_id = fields.Many2one(
+        'hr.job',
+        string="Job Title",
+        compute="_compute_main_job_position",
+        store=True)
+
+    @api.depends('contract_job_ids')
+    @api.multi
+    def _compute_main_job_position(self):
+        """
+        Get the main job position from the field contract_job_ids which
+        contains one and only one record with field is_main_job == True
+        """
+        for contract in self:
+            main_job = contract.contract_job_ids.filtered('is_main_job'
+                                                          ) or False
```

Comment:
You can remove `or False` because the filtered method will return an empty recordset if no record matches the condition. An empty recordset will be evaluated as False in the following conditional statement.

Reply:
@espo-tony Is not need to remove, because after @Trivedi-Vacha-SerpentCS  check `if condition`